etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java

```diff
@@ -0,0 +1,141 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.context.ApplicationContext;
+import gov.hhs.cdc.trustedintermediary.etor.messages.IdentifierCode;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.StringType;
+
+public class MapLocalObservationCodes implements CustomFhirTransformation {
+    protected final Logger logger = ApplicationContext.getImplementation(Logger.class);
+
+    private HashMap<String, IdentifierCode> map;
+
+    public MapLocalObservationCodes() {
+        InitMap();
+    }
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+        var observations = HapiHelper.resourcesInBundle(bundle, Observation.class);
+
+        for (Observation obv : observations.toList()) {
+            var codingList = obv.getCode().getCoding();
+            var updatedList = new ArrayList<Coding>();
+
+            for (Coding coding : codingList) {
```

Comment:
I think we may want to reconsider the use of a for loop here. When we map a local code, there should be only a single coding in this list and it will have an "alt-coding" extension. If we check for that here before proceeding, that should fix the case discussed in the comments below, and gets rid of that corner case issue.